Week3/homework/6-step3.js

```diff
@@ -4,3 +4,11 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 // add your solution here, or add a comment on how you would tackle this problem
+
+for ( var i = 0; i < arr.length; i++) {
+    for ( var j = 0; j < arr[i].length; j++) {
+        for ( var k = 0; k < arr[j].length; k++) {
+            product *= arr[i][j][k];
+        }
```

Comment:
Aahaan. I think this works just fine. However, what if I had a 4d array or a 5d array. Our code would break under such scenarios. Such scenarios would be best handled by a pattern in programming which we call **recursion**. I would suggest that you give it a read on the internet. Briefly, recursion means calling the same function over and over again until it satisfies a condition. 🙂